targets/unity-v2/source/Shared/Public/Pipeline/EventBatchingStage.cs

```diff
@@ -0,0 +1,172 @@
+                        this.stopwatch.Reset();
+                    }
+                }
+            }
+            else
+            {
+                logger.Error($"Event {eventRequest.Event.Name} failed validation check and was ignored");
+            }
+        }
+
+        protected override IEnumerable<IPlayFabEmitEventRequest> GetInputConsumingEnumerable()
+        {
+            IPlayFabEmitEventRequest eventRequest;
+            bool isEventObtained;
+            while (!this.input.IsCompleted)
+            {
+                // Attempt to get an event from its buffer, wait no longer than a small amount of time
+                isEventObtained = this.input.TryTake(out eventRequest, EventAvailabilityCheckTimeoutInMs, cts.Token);
+                if (cts.Token.IsCancellationRequested)
+                {
+                    // Exit enumerator if cancellation was signaled
+                    break;
+                }
+
+                if (stopwatch.Elapsed > this.BatchFillTimeout)
+                {
+                    // It is time to send the batches
+                    this.StoreAllBatches();
+                }
+
+                if (isEventObtained)
+                {
+                    yield return eventRequest;
+                }
+            }
+
+            // Done iterating through input collection
+            // but we may have something left in the batch
+            if (!cts.Token.IsCancellationRequested)
+            {
+                this.StoreAllBatches();
+            }
+
+            this.stopwatch.Stop();
+        }
+
+        private void StoreBatch(string titleId)
+        {
+            // Store a batch (and create a new one) only if it is not empty
+            if (this.batches[titleId].Count > 0)
+            {
+                this.StoreOutput(new TitleEventBatch(titleId, this.batches[titleId]));
+            }
+        }
+
+        private void StoreAllBatches()
+        {
+            foreach (var kvp in this.batches)
+            {
+                this.StoreBatch(kvp.Key);
+            }
+
+            //Clear all the batches that have been sent
+            this.batches.Clear();
+
+            // Stop keeping track of batch's age when it is stored
+            this.stopwatch.Reset();
+        }
+
+        private void CreateNewBatch(string titleId)
+        {
+            this.batches[titleId] = new List<IPlayFabEmitEventRequest>(this.BatchSize);
+        }
+
+        /// <summary>
+        /// Attempts to initialize a batch. Will no-op if batch already exists
+        /// </summary>
+        /// <param name="titleId"></param>
+        private void InitNewBatch(string titleId)
+        {
+            if (!this.batches.ContainsKey(titleId))
+            {
+                this.CreateNewBatch(titleId);
+            }
+        }
+
+        private static bool ValidationCheck(IPlayFabEmitEventRequest request)
+        {
+            // Payload size validation
+            // TODO
```

Comment:
This seems important.

Reply:
It was reserved for the future. Please keep it as is for now.

---
In reply to: [264019975](https://github.com/PlayFab/SDKGenerator/pull/436#discussion_r264019975) [](ancestors = 264019975)